avogadro/core/molecule.cpp

```diff
@@ -377,6 +392,8 @@ bool Molecule::removeAtom(Index index)
     m_hybridizations.pop_back();
   if (m_formalCharges.size() == m_atomicNumbers.size())
     m_formalCharges.pop_back();
+  if (m_colors.size() == m_colors.size())
```

Comment:
I think you meant `if (m_colors.size() == m_atomicNumbers.size())` here (same as above)